packages/services/schema/src/api.ts

```diff
@@ -160,23 +159,25 @@ export const schemaBuilderApiRouter = t.router({
         assertAllCasesExhausted(input);
       } catch (error) {
         // Treat timeouts caused by external composition as "expected errors"
-        if (ctx.cache.isTimeoutError(error) && input.type === 'federation' && input.external) {
+        if (ctx.cache.isTimeoutError(error)) {
           return {
             errors: [
               {
-                message: error.message,
-                source: 'graphql',
+                message: 'The composition timed out. Please try again later.',
+                source: 'composition',
               },
             ],
             sdl: null,
             supergraph: null,
-            includesNetworkError: true,
+            includesNetworkError: false,
+            includesException: true,
```

Comment:
This is not a network error but an exception because this basically means that we either
1. have an infinite loop in our composition logic
2. have external composition timeouts configured so they do not time out before external composition calls time out